### ISSUE TITLE
"Share room" should share bare URL

### ISSUE THREAD

**Issue by opusforlife2** [vector-im/element-web#17498]
#### Is your suggestion related to a problem? Please describe.

When using the Share room feature, the URL displayed contains several `via` parameters, such as `via=matrix.org`. This exposes the homeserver of the user as well, which may be needless (private) extra information if you're sharing a link to a public Room.

#### Describe the solution you'd like.

The user should have the (default) option to share the bare URL, without any parameters.

**Comment by aaronraimist** [comment]
They are not needless if you are sharing a room ID. The via parameters must be there for the link to work properly for a user who has never joined that room before.

There is a bug where the via parameters are also being added when you share a that has a human readable room alias. When you are sharing a room with an alias they are unnecessary. https://github.com/matrix-org/matrix-react-sdk/pull/6118 fixes that bug.

**Comment by t3chguy** [comment]
Also the via's are the most prominent servers in the room, not necessarily your server

**Comment by opusforlife2** [comment]
Ah, yes. I am indeed sharing an IRC room (https://matrix.to/#/#newpipe:libera.chat). That is an alias, right?

**Comment by aaronraimist** [comment]
Yes that's an alias

**Comment by opusforlife2** [comment]
> most prominent servers in the room

What is the threshold for prominence?

**Comment by t3chguy** [comment]
User count

**Comment by opusforlife2** [comment]
N decided by the homeserver the Room is hosted on? Or the HS of the sharing user?

**Comment by aaronraimist** [comment]
The room isn't hosted on a single server. Every server that participates in the room has a copy of the room and knows everyone that is in it.

**Comment by opusforlife2** [comment]
I'm aware every server has its own copy. But there is no such thing as a "master" copy?

**Comment by t3chguy** [comment]
No, there is no "master" server. All servers are effectively equal in the room, the power comes from the privileged users in the room. The servers with most users are chosen for vias as the chances of that server still being in the room at the time of the join bootstrap

**Comment by aaronraimist** [comment]
No. Every server has an equal copy. No server owns the room or has any special privileges or information.

**Comment by t3chguy** [comment]
The number of via's to use is up to the client, I believe Web takes the top 3

**Comment by opusforlife2** [comment]
> No server owns the room or has any special privileges or information.

Is this the case for IRC bridged Rooms as well? I thought only matrix.org has the bridge connection to Libera Chat.

Sorry, this is off-topic, but I'm curious.

**Comment by t3chguy** [comment]
Sure, the bridging itself is done by matrix.org - but even if matrix.org dies, all Matrix non-matrix.org users would still be able to communicate in that room and that room continues to function

**Comment by opusforlife2** [comment]
Wow, that is wild! I'm amazed you guys were able to make it work that way.
